src/definitions_cxx.hpp

```diff
@@ -565,6 +565,18 @@ enum class Error {
 	SD_CARD_NO_FILESYSTEM,
 };
 
+#define D_TRY(expr)                                                                                                    \
+	error = (expr);                                                                                                    \
+	if (error != Error::NONE) {                                                                                        \
+		return error;                                                                                                  \
+	}
```

Comment:
For this version, again, it's a little strange to have the macro depend on Error error being in scope. What would you think of:
```
#define D_TRY(expr)                              \
do {                               \
	Error error = (expr);                                              \
	if (error != Error::NONE) {                                         \
		return error;                                                      \
	}                          \
} while(0);
```